src/main/java/com/sonyericsson/jenkins/plugins/bfa/db/MongoDBKnowledgeBase.java

```diff
@@ -709,17 +709,23 @@ public Descriptor<KnowledgeBase> getDescriptor() {
      * Gets the connection to the MongoDB
      * @return the Mongo.
      */
-    private MongoClient getMongoConnection() {
+    MongoClient getMongoConnection() {
         if (mongo == null) {
             StringBuilder connectionStringBuilder = new StringBuilder(host);
             connectionStringBuilder.append(":").append(port);
             MongoClientSettings.Builder builder = MongoClientSettings.builder().applyToClusterSettings(
                     builder1 -> {
-                        List<ServerAddress> hostlist = new LinkedList<ServerAddress>();
-                        hostlist.add(new ServerAddress(host, port));
+                        List<ServerAddress> hostlist = new LinkedList<>();
+                        for (String h: host.split(",")) {
+                            hostlist.add(new ServerAddress(h, port));
```

Comment:
Some input sanitation seems to be needed here. For example:
```suggestion
                            h = StringUtils.trimToNull(h);
                            if (h != null) {
                                hostlist.add(new ServerAddress(h, port));
                            }
```

A future improvement would be to allow each host to have a port specified as well `host1:port,hots2:port,...`

Reply:
> Some input sanitation seems to be needed here. For example:
> 
> A future improvement could be to allow each host to have a port specified as well `host1:port,hots2:port,...`

I concur, that is a good future improvement.